### PR TITLE
Support the 3D Connexion space mice.

### PR DESCRIPTION
**Pull request by SirVer**

### CODE REVIEW

**Comment by SirVer** [comment]
meow @tblaette

File: sdl_viewer/src/lib.rs [hunk 1]
```diff
@@ -522,6 +533,17 @@ impl SdlViewer {
                 }
             }
 
+            if let Some(j) = joystick.as_ref() {
```
Comment (tblaette) [inline sdl_viewer/src/lib.rs]: what is the `.as_ref()` for?

Comment (SirVer) [reply sdl_viewer/src/lib.rs]: Joystick is a Some(Joystick). If I try to pattern match, the compiler tries to move it out of the Option. I do not want that, I just want a reference to it and keep it inside the Option. hence the as_ref().

**Comment by SirVer** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user SirVer. Merge queue now has a length of 1.

**Pull request merged**